### PR TITLE
bpo-43655: Tkinter and IDLE dialog windows are now recognized as dialogs by window managers on macOS and X Window

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue43655

### CODE REVIEW

File: Lib/idlelib/configdialog.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 from tkinter import colorchooser
 import tkinter.font as tkfont
 from tkinter import messagebox
+from tkinter.simpledialog import _setup_dialog
```
Comment (terryjreedy) [inline Lib/idlelib/configdialog.py]: As explained on the issue, I don't want this dialog changed on mac.  I don't know what happens on Linux.

File: Lib/tkinter/filedialog.py [hunk 2]
```diff
@@ -62,6 +62,7 @@ def __init__(self, master, title=None):
         self.top = Toplevel(master)
         self.top.title(title)
         self.top.iconname(title)
+        _setup_dialog(self.top)
```
Comment (E-Paine) [inline Lib/tkinter/filedialog.py:66]: This needs to be imported

**Review by E-Paine — COMMENTED** [review]
IDLE dialogs and tkinter simpledialogs fine. However, even after adding the import filedialog does not behave as expected as it does not close when the escape key is pressed (all the others do). I haven't been able to find the reason for the difference between simpledialog and filedialog yet.

Tested on MacOS (my Linux setup handled modal dialogs correctly before this patch)

**EDIT:**
Also, would it make more sense for `_setup_dialog` to be in `tkinter.dialog` instead?

File: Lib/tkinter/filedialog.py (hunk 2 above)
Comment (serhiy-storchaka) [reply Lib/tkinter/filedialog.py:66]: Good catch!

File: Lib/idlelib/configdialog.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/idlelib/configdialog.py]: Done.

**Comment by serhiy-storchaka** [comment]
> Also, would it make more sense for `_setup_dialog` to be in `tkinter.dialog` instead?

I already added helper `_place_window` in `tkinter.simpledialog`, so will add a new helper here too.

There is no "main" dialog file. `tkinter.dialog`, `tkinter.simpledialog` and `tkinter.commondialog` define different "dialog" classes which have almost nothing in common. Maybe after rewriting dialog implementations I make them sharing more code and move helpers to more appropriate place. Maybe even make some of helpers public.

**Comment by E-Paine** [comment]
> I haven't been able to find the reason for the difference between simpledialog and filedialog yet.

Oops, ignore me! `simpledialog` binds the "<Escape>" event to `cancel` (I presume IDLE does similar). LGTM.

**Pull request merged**